doc/motr-developer-guide.md

```diff
@@ -0,0 +1,235 @@
+features.
+
+The first document developers should read is: [Motr Client API ](/motr/client.h).
+It explains basic notations, terminologies, and data structures in Cortx Motr.
+Developers are also assumed having a running Cortx Motr system: please refer to
+[Cluster Setup](https://github.com/Seagate/Cortx/blob/main/doc/Cluster_Setup.md)
+and [Quick Start Guide](/doc/Quick-Start-Guide.rst)
+
+Cortx Motr provides object based operations and index (a.k.a key/value) based operations.
+
+## A simple Cortx Motr application (object)
+
+Here we will create a simple Cortx Motr application to crate an object, write
+some data to this object, read data back from this object, and then delete it.
+Source code is available at: [example1.c](/motr/examples/example1.c)
+
+  * checkout the source code, make, run unit test
+  
+	$ git clone --recursive git@github.com:Seagate/cortx-motr.git -b main
+	
+	$ cd cortx-motr
+	
+	$ sudo ./scripts/m0 make
+	
+	$ sudo ./scripts/m0 run-ut
+	
+
+  * Create a source code file, and start programming.
+  
+	$ cd your_project_dir
+
+	$ touch example1.c
+
+  * Include necessary header files
+```C	
+	#include "motr/client.h"
+```
+There are also various Motr libraries header files that can be used in
+Motr applications. Please refer to source code "lib/"
+
+  * Define necessary variables (global or in main() function)
+
+```C
+	static struct m0_client         *m0_instance = NULL;
+	static struct m0_container       motr_container;
+	static struct m0_config          motr_conf;
+	static struct m0_idx_dix_config  motr_dix_conf;
+```
+
+  * Get configuration arguments from command line
+```C
+
+	#include "motr/client.h"
+
+	static struct m0_client         *m0_instance = NULL;
+	static struct m0_container       motr_container;
+	static struct m0_config          motr_conf;
+	static struct m0_idx_dix_config  motr_dix_conf;
+
+	int main(int argc, char *argv[])
+	{
+	        int rc;
+
+	        motr_dix_conf.kc_create_meta = false;
+
+	        motr_conf.mc_is_oostore            = true;
+	        motr_conf.mc_is_read_verify        = false;
+	        motr_conf.mc_ha_addr               = argv[1];
+	        motr_conf.mc_local_addr            = argv[2];
+	        motr_conf.mc_profile               = argv[3];
+	        motr_conf.mc_process_fid           = argv[4];
+	        motr_conf.mc_tm_recv_queue_min_len = M0_NET_TM_RECV_QUEUE_DEF_LEN;
+	        motr_conf.mc_max_rpc_msg_size      = M0_RPC_DEF_MAX_RPC_MSG_SIZE;
+	        motr_conf.mc_idx_service_id        = M0_IDX_DIX;
+	        motr_conf.mc_idx_service_conf      = (void *)&motr_dix_conf;
+
+	        return 0;
+	}
+```
+The most important and necessary configuration parameters:
+
+    - HA Addr    : HA service addr in Motr service
+    - Local Addr : Local addr used to connect to Motr service
+    - Profile FID: The Profile FID in Motr service
+    - Process FID: The Process FID in Motr service
+
+These parameters can be queried from:
+
+    - `hctl status` from a Motr depolyment, or
```

Comment:
hctl command will be available after hare source compilation. Here we are using motr source code. I believe, no need for the hctl command here.

```
-bash-4.2$ which hctl
/usr/bin/hctl
-bash-4.2$ ls -l /usr/bin/hctl
lrwxrwxrwx 1 root root 42 Oct 26 23:26 /usr/bin/hctl -> /home/744290/src/centos7.8/cortx-hare/hctl
-bash-4.2$
```

Reply:
So, I proposed two ways to get these parameters:
1. from a Motr deployment. Maybe I should say, a Cortx Motr deployment?
    But the developers won't need to deploy all the components from Cortx. Cortx-motr, and Cortx-hare is enough. This is described in these two document:
    [Cluster Setup](https://github.com/Seagate/Cortx/blob/main/doc/Cluster_Setup.md)
and [Quick Start Guide](/doc/Quick-Start-Guide.rst)

2. Developer can run a script to setup a single node Motr. This is described as a second option below.

But as you pointed out, it's better to add more details here.